### ISSUE TITLE
Please let load_weights load only weights as its name describes

### ISSUE THREAD

**Issue by netw0rkf10w** [tensorflow/tensorflow#34036]
**System information**
- TensorFlow version (you are using): 2.0.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current confusing/bad behavior/state.**

Look at the end of [this tutorial](https://www.tensorflow.org/guide/keras/save_and_serialize):

```
# Recreate the model
new_model = get_model()
new_model.compile(loss='sparse_categorical_crossentropy',
                  optimizer=keras.optimizers.RMSprop())

# This initializes the variables used by the optimizers,
# as well as any stateful metric variables
new_model.train_on_batch(x_train[:1], y_train[:1])

# Load the state of the old model
new_model.load_weights('path_to_my_weights')

# Check that the model state has been preserved
new_predictions = new_model.predict(x_test)
np.testing.assert_allclose(predictions, new_predictions, rtol=1e-6, atol=1e-6)

# The optimizer state is preserved as well,
# so you can resume training where you left off
new_first_batch_loss = new_model.train_on_batch(x_train[:64], y_train[:64])
assert first_batch_loss == new_first_batch_loss
```

Look more closely:

```
# Load the state of the old model
new_model.load_weights('path_to_my_weights')
# The optimizer state is preserved as well
```

I'm sorry, but, are you serious? Why on earth do you think it's a good idea to let a **load_weights** function restore anything other than the **WEIGHTS** of the model? Is this deep learning from 3019? Maybe it's a documentation issue?

If you mean to give the users the option to save or load other variables in addition to the model's weights, please create new functions, e.g. **save_variables()** and **load_variables()**.

**Will this change the current api? How?**
Yes, in a good way.

**Who will benefit with this feature?**
Everybody.

**Comment by rmothukuru** [comment]
@netw0rkf10w,
Sorry for the delayed response. 

1. Can you please let us know the reference which states that `load_weights` preserves the **`Optimizer`** state as well.
2. Even if if it is so, please let us know the disadvantage of that functionality.

**Comment by netw0rkf10w** [comment]
Hi @rmothukuru.

> 1. Can you please let us know the reference which states that `load_weights` preserves the **`Optimizer`** state as well.

I no longer use TensorFlow, but isn't this clear from the description that I posted above? (Look at this comment in the tutorial: `The optimizer state is preserved as well`.)

> 2. Even if if it is so, please let us know the disadvantage of that functionality.

The issue here is not with the functionality, but a mismatch between the name of a function and what it does. **It is confusing**. If I use a `load_weights` function, I'd expect it to only load weights, nothing else (by default).

It's fine if this is not an issue, TensorFlow users get confused all the time anyway.

**Comment by rmothukuru** [comment]
@netw0rkf10w,
Thank you for your response. It's been more than an year that you raised this issue and no other Developer in the community expressed their **confusion** regarding this, Surprisingly. So, closing this issue as it is neither a Bug nor Feature.

**Comment by netw0rkf10w** [comment]
@rmothukuru Note that the behavior of the `load_weights` function has been changed (and the above tutorial has also been largely updated). If it was good then it wouldn't have been changed, so the issue I created was actually very relevant at that time (i.e., nearly two years ago).

Your replies above show that you were confused when reading this issue, which is because you take nearly two years to read it.
